File: Ruby/euler21.rb
#!/usr/bin/env ruby

class Integer
  def sum_divisors
    sum = 0
    (1..(self / 2)).each do |i|
      sum += i if self % i == 0
    end
    sum
  end
end


sum = 0
(1..10_000).each do |i|
  sumd = i.sum_divisors
  sum += i if sumd.sum_divisors == i && sumd != i
end

puts sum

File: C/euler21.c
#include <stdio.h>
#include <math.h>

int sum_divisors(int num)
{
    int sum=1;
    int i;
    for(i=2; i*i <= num; sum += num % i ? 0 : (i*i == num ? i : i + num/i) , ++i);
    return sum;
}

int main()
{
	int sum = 0;
	int i;
	int temp = 0;
	for(i = 1; i < 10000; ++i)
	{
		temp = sum_divisors(i);
		if(sum_divisors(temp) == i && temp != i)
			sum += i;
	}
	printf("%d\n", sum);

	return 0;
}
